src/main/java/commands/DeleteMangaCommand.java

```diff
@@ -0,0 +1,86 @@
+
+import static constants.Command.AUTHOR_INDEX;
+import static constants.Command.DELETE_COMMAND;
+import static constants.Command.MANGA_INDEX;
+import static storage.StorageHelper.saveFile;
+
+//@@author sarahchow03
+/**
+ * Represents a command to delete a manga from an author's mangaList.
+ * It checks for the existence of the author and manga in the list before deletion.
+ * If the author or manga is not found, an exception is thrown.
+ *
+ * This class extends the Command class and overrides the execute method to handle the delete operation.
+ */
+public class DeleteMangaCommand extends Command {
+    // private static final Logger logger = Logger.getLogger(DeleteMangaCommand.class.getName());
+    private String authorName;
+    private String mangaName;
+
+
+    /**
+     * Constructs a DeleteMangaCommand with the given user input.
+     *
+     * @param userInputList The user's input list which consists of the author and manga.
+     */
+    public DeleteMangaCommand(String[] argsAuthorManga) {
+        super(DELETE_COMMAND);
+        this.authorName = argsAuthorManga[AUTHOR_INDEX];
+        this.mangaName = argsAuthorManga[MANGA_INDEX];
+    }
+
+    /**
+     * Executes the delete command by conducting several checks on the user input and author list.
+     * If the checks are successful, the specified manga is deleted from the author's list.
+     *
+     * @param ui          The Ui object that handles user interface interactions.
+     * @param authorList  The list of authors containing the list of existing authors in the catalog.
+     * @throws TantouException If the author or manga does not exist, or if the input is invalid.
+     */
+    @Override
+    public void execute(Ui ui, AuthorList authorList) throws TantouException {
+        if (authorName.isEmpty() || mangaName.isEmpty()) {
+            logger.log(Level.SEVERE, "Author name or manga name is empty");
+            throw new TantouException("No author or manga provided!");
+        }
+        assert (!authorName.isEmpty() && !mangaName.isEmpty()) : "Author or manga name is empty";
+        logger.log(Level.INFO, "Deleting manga... " + mangaName + " from " + authorName);
+
+        Author attachedAuthor = new Author(authorName);
+        Manga deletingManga = new Manga(mangaName, attachedAuthor);
+
+        if (authorList.hasAuthor(attachedAuthor)) {
+            Author existingAuthor = authorList.getAuthor(attachedAuthor);
+            if (existingAuthor.hasManga(deletingManga)) {
+                existingAuthor.deleteManga(deletingManga);
+                System.out.printf("Manga %s successfully deleted from author %s\n",
+                        deletingManga.getMangaName(), existingAuthor.getAuthorName());
+                logger.log(Level.INFO, "Successfully deleted manga: " + deletingManga.getMangaName());
+
+                saveFile(authorList);
+                return;
+            }
+            assert !existingAuthor.hasManga(deletingManga): "No manga found";
```

Comment:
The assertion check for finding the existence of authorName after already validating it before could be removed since they are unnecessary